File: solutions/csharp/MinStack.cs
public class MinStack
{
    private Node head;

    public MinStack()
    {
        head = null;
    }

    public void Push()
    {

    }

    public void Pop()
    {

    }

    public int Top()
    {

    }

    public int GetMin()
    {

    }

    private class Node
    {
        public Node(int value, int minValue, Node next = null)
        {
            Value = value;
            MinValue = minValue;
            Next = next;
        }

        public int Value { get; set; }
        public int MinValue { get; set; }
        public Node Next { get; set; }
    }
}

File: solutions/java/minStack.java
class MinStack {
	Node head;

	public MinStack() {
		head = null;
	}

	public void push(int val) {
		if (head == null) {
			head = new Node(val, val, head);
		} else {
			head = new Node(val, Math.min(val, head.minVal), head);
		}

	}

	public void pop() {
		head = head.next;
	}

	public int top() {
		return head.val;
	}

	public int getMin() {
		return head.minVal;
	}

}

class Node {
	public int val;
	public int minVal;
	public Node next;

	public Node(int val, int minVal, Node next) {
		this.val = val;
		this.minVal = minVal;
		this.next = next;
	}

	public Node(int val, int minVal) {
		this(val, minVal, null);
	}
}
